model/protecode.py

```diff
@@ -12,13 +12,19 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import enum
 
 from model.base import (
-    BasicCredentials,
+    BasicCredentials, ModelValidationError,
     NamedModelElement,
 )
 
 
+class ProtecodeAuthScheme(enum.Enum):
+    BASIC_AUTH = 'basic_auth'
+    OAUTH = 'oauth'
```

Comment:
nit: are you sure that the (new) auth-scheme we use against protecode is actually oauth? I think, oauth would include the actual token-retrieval (which I understood from our last discussion we do upfront); so maybe `token` or `token_auth`, or `bearer_token` might be more adequate names